pkg/rehearse/jobs_test.go

```diff
@@ -1454,3 +1454,64 @@ func TestDetermineJobURLPrefix(t *testing.T) {
 		})
 	}
 }
+
+func TestSortConfigs(t *testing.T) {
+	testCases := []struct {
+		name     string
+		configs  []*config.DataWithInfo
+		expected []*config.DataWithInfo
+	}{
+		{
+			name: "2 configs",
```

Comment:
Let's add more tests that prepare us for the task at hand. I think we need the following tests:

- same org/repo, branches `main` and `release-4.10`
- different org/repo, branches `main` and `release-4.10`
- same org/repo, branches `release-4.9` and `release-4.10`
- different org/repo, branches `release-4.9` and `release-4.10`
- same org/repo, branches `master` and something else than both master and release branch
- same org/repo, branches `release-4.1` and `release-4.10` 

Having these tests will allow you show how exactly does the behavior change in the final fix PR.